subsys/nrf_security/Kconfig.legacy

```diff
@@ -76,81 +76,82 @@ config MBEDTLS_AES_DECRYPT_ALT
 
 config MBEDTLS_AES_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_CIPHER_DRIVER
 	default y
 
 config MBEDTLS_CMAC_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_CMAC
 	default y
 
 config MBEDTLS_CCM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_AES_CCM
 	default y
 
 config MBEDTLS_GCM_ALT
 	bool
 	depends on CC312_BACKEND || \
-		   (PSA_CRYPTO_DRIVER_CC3XX && CRYPTOCELL_CC312_USABLE)
+		   (PSA_NEED_CC3XX_AES_GCM && CRYPTOCELL_CC312_USABLE)
 	default y
 
 config MBEDTLS_CHACHA20_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_NEED_OBERON_CHACHA20 || \
+		   CC3XX_BACKEND || PSA_NEED_CC3XX_CHACHA20
 	default y
 
 config MBEDTLS_POLY1305_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || PSA_NEED_OBERON_CHACHA20_POLY1305 || \
+		   CC3XX_BACKEND || PSA_NEED_CC3XX_CHACHA20_POLY1305
 	default y
 
 config MBEDTLS_CHACHAPOLY_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || PSA_NEED_CC3XX_CHACHA20_POLY1305
```

Comment:
Don't we need the same dependencies as the MBEDTLS_POLY1305_ALT here?
Meaning we don't have Oberon the same way as we have it there.